laser.py

```diff
@@ -130,16 +133,30 @@ def stopLaser( self ):
   def configureScanDataOutput( self ):
     pass
 
+
+def write_timestamp(f):
+  """write time.time() in sec/frac into a binary file"""
+  t = time.time()
+  sec = int(t)
+  frac = int(0x10000*(t-sec))
+  f.write(struct.pack('IH', sec, frac))
```

Comment:
I don't like floats :worried:

Reply:
I know that you do not like floats, but I do not remember how you wanted to handle time.time(). My motivation was that seconds are too coarse and if I use milliseconds, then I need 2 bytes -> I can use 1^-16 fraction instead.

Reply:
I don't really care that much about the number of bytes used. What I really don't like is, that you start with float, convert it to int, write it to log, read it and convert it back to float.

As for the sizes:
* 'IH' is 6 bytes - epoch starts at 1970
* [NTP uses 64bit](https://en.wikipedia.org/wiki/Network_Time_Protocol#Timestamps) - epoch starts at 1900
* Python [timedelta uses (days, seconds, micro)](https://docs.python.org/2/library/datetime.html?highlight=timedelta#datetime.timedelta) - you can pick your epoch - I'd pick the start of the log

To summarize: 
* I'd use timedelta for all calculations in the program
* I'd let you pick what's written in the log (but why not use NTP timestamps?)
* I'd log the absolute time of the start of the log at the start of the log (the absolute time is usually not needed for anything so this is "just in case")

Reply:
But I don't care that much. Feel free to keep you implementation. However, it would be great to agree on a way to handle all time aspects for the future.

Reply:
Another option is to use your Python delta from the beginning of recording in microseconds (4 bytes), which means
```
>>> 0x100000000/1000000.
4294.967296
```
seconds, i.e. more than 1hour of recording. If the log file contains video or Velodyne data you should split it into multiple files anyway. Also overflow should be clearly visible.

Reply:
Alternative attempt is implemented in https://github.com/robotika/naio/pull/1